raiden/raiden_service.py

```diff
@@ -659,13 +657,12 @@ def _initialize_wal(self) -> None:
                 storage=storage, node_address=self.address, initial_state=initial_state
             )
 
-            unapplied_state_changes_range = Range(state_change_start, HIGH_STATECHANGE_ULID)
-            state_change_qty_unapplied, snapshot_state = wal.replay_unapplied_state_changes(
-                transition_function=node.state_transition,
-                storage=storage,
-                unapplied_state_changes_range=unapplied_state_changes_range,
+            state, state_change_qty_unapplied = wal.replay_state_changes(
                 node_address=self.address,
-                state_snapshot=state_snapshot,
+                state=state_snapshot,
+                state_change_range=Range(state_change_start, HIGH_STATECHANGE_ULID),
+                storage=storage,
+                transition_function=node.state_transition,  # type: ignore
```

Comment:
Without the ignore this yields:
```
raiden/raiden_service.py:665: error: Argument "transition_function" to "replay_state_changes" has incompatible type "Callable[[ChainState, StateChange], TransitionResult[ChainState]]"; expected "Callable[[State, StateChange], TransitionResult[State]]"
```
Any ideas how to fix this?

Reply:
You should be able to fix it by making the state parameter and first return value of `replay_state_changes` generic. You can bind your generic variable to `State` to only accept `State` subtypes.

Is `replay_state_changes` used on other states than `ChainState`? if not, you could just used declare that param as a `ChainState` instead of using generics.